include/chowdsp_wdf/rtype/rtype_detail.h

```diff
@@ -67,7 +67,7 @@ namespace wdft
                 v_type bc {};
                 for (int r = 0; r < vec_size; r += simd_size)
                     bc = xsimd::fma (xsimd::load_aligned (S_[c].data() + r), xsimd::load_aligned (a_ + r), bc);
-                b_[c] = xsimd::hadd (bc);
+                b_[c] = xsimd::reduce_add (bc);
```

Comment:
Oh nice, this is definitely a good idea! My only worry is that this is a breaking change for folks using an older version of XSIMD (this is also why the CI for this pull request is currently failing). Probably a more backwards-compatible solution would be to do something like:
```cpp
#if XSIMD_VERSION >= <version where reduce_add was added>
    b_[c] = xsimd::reduce_add (bc);
#else
    b_[c] = xsimd::hadd (bc);
#endif
```
I'm not exactly sure how XSIMD expresses the library version in their code, but I'm guessing it's something along those lines.

It would also be cool to leave a comment for why `xsimd::reduce_add` is preferred. E.g., does it have better performance? was `xsimd::hadd()` deprecated?

This is also a good reminder for me to make sure the documentation is clear about which versions of XSIMD are supported, which should also including expanding the CI setup to test the library with multiple version of XSIMD. I'll take care of that in a separate PR :).

Reply:
Yes, good point about backwards-compatibility. I'll poke around and see if there is any way to check for the XSIMD version of the library.

Regarding the change, I'm not exactly sure what the reasoning was behind it, but it looks like they just swapped out hadd() for reduce_add()--my guess is that the new function name is more clear as to what it actually does.

Reply:
Yeah, I just noticed that as well. I was able to add CI tests for different versions of XSIMD in #6, and it looks like the current implementation compiles successfully with the latest tagged release.

Because of that, doing a macro check like we had talked about won't work until a new version is released, since the [current version macros](https://github.com/xtensor-stack/xsimd/blob/master/include/xsimd/config/xsimd_config.hpp#L15) are still showing version 8.1.0.

Maybe the thing to do then is to wait for the next tagged release? Or maybe there's some other workaround.

Reply:
Yeah, waiting for the next release is probably the easiest route. I submitted a [new issue](https://github.com/xtensor-stack/xsimd/issues/802) over at xsimd to see if they'll bump the current version number.